include/highfive/bits/H5Converter_misc.hpp

```diff
@@ -628,16 +636,31 @@ struct inspector<boost::numeric::ublas::matrix<T>> {
 #endif
 
 template <typename T>
-class data_converter {
-    static Reader<T> get_readers(const std::vector<size_t> dims) {
-        size_t n_dim = inspector<T>::recursive_ndim;
-        if (n_dim < dims.size()) {
-            throw std::string("Error dataset is too larger for T");
-        } else if (n_dim > dims.size()) {
+struct data_converter {
+    static std::vector<size_t> try_reduce_dims(const std::vector<size_t> dims, size_t n) {
+        std::vector<size_t> dims_{};
+        for (auto it = dims.crbegin(); it != dims.crend() && dims_.size() != n; ++it) {
```

Comment:
With `dims_.size() != n` in the loop condition, it seems to me that the resulting `dims_` will always be `<= n`.

But below you then test `dims_.size() > n`, I don't see how that will ever be triggered?

Reply:
If we try to fit {3, 3, 1} into 1D it will not works.
And the loop will stop after the last element.

Reply:
Do we have a test for that?

Reply:
https://github.com/BlueBrain/HighFive/blob/master/tests/unit/tests_high_five_base.cpp#L1533-L1569

Reply:
Hm, tried to cover that condition in there, and got told off by https://github.com/BlueBrain/HighFive/blob/start_serialize/include/highfive/bits/H5Dataspace_misc.hpp#L145

I'm still wondering how to cover it, but I think that most of our use cases are kept in check, too.

Reply:
I'd still feel more comfortable with it like this
```suggestion
        for (auto it = dims.crbegin(); it != dims.crend(); ++it) {
```
but maybe I'm overthinking this in a wrong way

Reply:
hm your patch looks wrong because we can go "too" far